arch/riscv/core/isr.S

```diff
@@ -443,12 +443,17 @@ do_fault:
 1:	mv a1, sp
 
 #ifdef CONFIG_EXCEPTION_DEBUG
+	/* Store csf addr in a2 to be passed as 3rd arg */
+	addi a2, sp, -__callee_saved_t_SIZEOF
+
+	/* Store csf addr to exception stack frame */
+	sr a2, __struct_arch_esf_csf_OFFSET(sp)
+
```

Comment:
Minor style suggestion: I think it's technically safe to store this pointer in the CSF before we've decremented sp, but in general I think it is cleaner to not reference not-yet-allocated stack space.  I think a more-obviously-safe option would be to revert back to the original code, then, right after `mv a2, sp` do a `sr a2 __struct_arch_esf_csf_OFFSET(a1)` to store the pointer.  (`a1` still holds the pointer to the esf at this point.)

Reply:
Thanks for pointing this out, I do find that my implementation is a little weird, this looks much cleaner.